src/util/dbid.cpp

```diff
@@ -0,0 +1,16 @@
+#include "util/dbid.h"
+
+
+const QVariant::Type DbId::kVariantType = DbId().toVariant().type();
+
+//static
+DbId::value_type DbId::valueOf(QVariant /*pass-by-value*/ variant) {
+    if (variant.canConvert<value_type>() &&
+            variant.convert(kVariantType)) {
+        value_type value = variant.value<value_type>();
```

Comment:
That was my first approach, but the unit tests revealed that canConvert() is 
not sufficient ;)

On 08/24/2015 02:51 PM, Daniel Schürmann wrote:
>
> In src/util/dbid.cpp 
> <https://github.com/mixxxdj/mixxx/pull/684#discussion_r37747090>:
>
> > @@ -0,0 +1,16 @@
> > +#include "util/dbid.h"
> > +
> > +
> > +const QVariant::Type DbId::kVariantType = DbId().toVariant().type();
> > +
> > +//static
> > +DbId::value_type DbId::valueOf(QVariant /*pass-by-value*/ variant) {
> > +    if (variant.canConvert<value_type>() &&
> > +            variant.convert(kVariantType)) {
> > +        value_type value = variant.value<value_type>();
>
> I have just digged though the QVariant source and it turns out that 
> convert() and value(); are calling both
> QVariant::handler->convert().
> The difference is that convert performs a in place conversion where 
> value() returns the value or the default constructor, which is 0 in our case.
> It seams that QVariant misses a all in one function that suit our needs.
>
> how about trust canConvert:
>
> |if (variant.canConvert(kVariantType) ) { value_type value = 
> variant.value<value_type>(); } |
>
> and use t const reference paramerter?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/684/files#r37747090>.
>